python/hail/dataset.py

```diff
@@ -2410,47 +2392,173 @@ def persist(self, storage_level="MEMORY_AND_DISK"):
             MEMORY_ONLY_SER_2, MEMORY_AND_DISK, MEMORY_AND_DISK_2,
             MEMORY_AND_DISK_SER, MEMORY_AND_DISK_SER_2, OFF_HEAP
 
-        :return: A VariantDataset that has been persisted.
-
-        :rtype: VariantDataset
-
         """
 
         pargs = ['persist']
         if storage_level:
             pargs.append('-s')
             pargs.append(storage_level)
-        return self.hc.run_command(self, pargs)
+        self.hc._run_command(self, pargs)
+
+    def global_schema(self):
+        """
+        Returns the signature of the global annotations contained in this VDS.
+
+        >>> print(vds.global_schema())
+
+        :rtype: :class:`.Type`
+        """
+        return Type._from_java(self._jvds.globalSignature())
+
+    def sample_schema(self):
+        """
+        Returns the signature of the sample annotations contained in this VDS.
+
+        >>> print(vds.sample_schema())
+
+        :rtype: :class:`.Type`
+        """
+
+        return Type._from_java(self._jvds.saSignature())
+
+    def variant_schema(self):
+        """
+        Returns the signature of the variant annotations contained in this VDS.
+
+        >>> print(vds.variant_schema())
+
+        :rtype: :class:`.Type`
+        """
+
+        return Type._from_java(self._jvds.vaSignature())
+
+    def query_samples_typed(self, exprs):
+        """Perform aggregation queries over samples and sample annotations, and returns python object(s) and types.
+
+        **Example**
+
+        >>> low_callrate_samples, t = vds.query_samples_typed(
+        >>>     'samples.filter(s => sa.qc.callRate < 0.95).collect()')
+
+        See :py:meth:`.query_samples` for more information.
+
+        :param exprs: one or more query expressions
+        :type exprs: str or list of str
+        :return: annotation(s) and type(s)
+        :rtype: tuple of annotation or list of annotation, and :class:`.Type` or list of :class:`.Type`
+        """
+
+        if (isinstance(exprs, list)):
```

Comment:
I'm confused by this.  Do you allow exprs to be a string with a comma-separated list of expresions?  Then I think this does the wrong thing.

I've never been a fan of this "if it is one you get the value, more you get a list."  I see the convenience generally it makes things more complicated, esp. if you want to use this in python code that builds pipelines programmatically (which everyone is doing).